File: job/src/main/java/com/job/wocus/service/JobService.java
package com.job.wocus.service;

import com.job.wocus.dto.*;

import java.util.List;

public interface JobService {
    JobDTO createJob(JobDTO jobDTO);

    List<JobDTO> getJobsByEmail(String email);

    List<JobDTO> getJobsByDomain(String domain);

    void deleteJob(Long id);

    JobDTO updateJob(Long id, JobDTO jobDTO);
}

File: wocus_react/src/services/jobService.js
import axios from "axios";

const BASE_URL = "http://localhost:8086/jobs";

const jobService = {
  createJob: (jobData) => {
    return axios.post(BASE_URL, jobData);
  },

  getJobsByEmail: (email) => {
    return axios.get(`${BASE_URL}/email/${email}`);
  },

  getJobsByDomain: (domain) => {
    return axios.get(`${BASE_URL}/domain/${domain}`);
  },

  deleteJob: (id) => {
    return axios.delete(`${BASE_URL}/${id}`);
  },

  updateJob: (id, jobData) => {
    return axios.put(`${BASE_URL}/${id}`, jobData);
  },
};

export default jobService;
